### PR TITLE
[Hotfix] - Store Detected Contour

### PR DESCRIPTION
**Pull request by FaaizHaikal**
## Description
Add method to store the detected contour

## Type of Change

- [x] Bugfix
- [ ] Enhancement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause the existing functionality to not work as expected)

## How Has This Been Tested?

- [ ] New unit tests added.
- [x] Manual tested.

## Checklist:

- [x] Using Branch Name Convention
    - `feature/JIRA-ID-SHORT-DESCRIPTION` if has a JIRA ticket
    - `enhancement/SHORT-DESCRIPTION` if has/has no JIRA ticket and contain enhancement
    - `hotfix/SHORT-DESCRIPTION` if the change doesn't need to be tested (urgent)
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made the documentation for the corresponding changes.

### CODE REVIEW

**Review by JayantiTA — APPROVED** [review]

File: include/ninshiki_cpp/utils/contours.hpp [hunk 1]
```diff
@@ -42,6 +42,7 @@ class Contours
 
   std::vector<std::vector<cv::Point>> get_contours();
   void set_contours(std::vector<std::vector<cv::Point>> contours);
+  void add_contour(std::vector<cv::Point> contour);
```
Comment (threeal) [inline include/ninshiki_cpp/utils/contours.hpp:45]:
```suggestion
  void add_contour(const std::vector<cv::Point>& contour);
```

**Review by threeal — CHANGES_REQUESTED** [review]
Anyway, since there are both `set_countours` and `add_countour` functions, i think it's better to just expose the `countours` variable instead.

File: include/ninshiki_cpp/utils/contours.hpp (hunk 1 above)
Comment (FaaizHaikal) [reply include/ninshiki_cpp/utils/contours.hpp:45]: You mean make it public, mas?

Comment (threeal) [reply include/ninshiki_cpp/utils/contours.hpp:45]: yes

File: include/ninshiki_cpp/utils/contours.hpp [hunk 2]
```diff
@@ -37,12 +37,12 @@ class Contours
...
   explicit Contours(cv::Mat binary_mat);
 
+  std::vector<std::vector<cv::Point>> contours;
+
   cv::Mat get_binary_mat(cv::Size mat_size);
   cv::Mat get_binary_mat_line(cv::Size mat_size, int line_size);
 
   std::vector<std::vector<cv::Point>> get_contours();
-  void set_contours(std::vector<std::vector<cv::Point>> contours);
-  void add_contour(std::vector<cv::Point> contour);
```
Comment (threeal) [inline include/ninshiki_cpp/utils/contours.hpp]: The `get_contours` function is unnecessary since the `contours` variable is already public.

**Review by threeal — APPROVED** [review]

File: data/color_classifier.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  },
+  "field": {
+    "min_hsv": [84, 10, 0],
+    "max_hsv": [116, 100, 100]
+  },
+  "line": {
+    "min_hsv": [180, 0, 55],
+    "max_hsv": [260, 60, 100]
+  }
+}
```
Comment (threeal) [inline data/color_classifier.json:14]: missing EoL

File: src/ninshiki_cpp/detector/color_detector.cpp [hunk 4]
```diff
@@ -62,9 +62,10 @@ bool ColorDetector::load_configuration(const std::string & path)
   nlohmann::json config = nlohmann::json::parse(input);
   for (auto & item : config.items()) {
     // Get all config
+    std::string name = item.key();
     try {
       utils::Color color(
-        item.value().at("name"),
+        name,
```
Comment (threeal) [inline src/ninshiki_cpp/detector/color_detector.cpp:68]: Straightforward:
```suggestion
        item.key(),
```

**Review by threeal — CHANGES_REQUESTED** [review]

**Review by threeal — APPROVED** [review]

**Pull request merged**